rotkehlchen/db/utils.py

```diff
@@ -363,3 +378,80 @@ def combine_asset_balances(balances: List[SingleDBAssetBalance]) -> List[SingleD
             new_balances = _append_or_combine(new_balances, balance)
 
     return new_balances
+
+
+def db_schema_sanity_check(
+        db_name: Literal['encrypted', 'global'],
+        cursor: 'DBCursor',
+        script_create_tables: str,
+) -> None:
+    """Ensures that database schema is not broken. Raises DBSchemaError if anything is bad."""
+    # Replacing new lines and white spaces since they may vary if by an accident code of a
+    # db upgrade was a bit different from the one that creates new tables
+    create_script_from_constant = script_create_tables.replace(' ', '').replace('\n', '').replace('\'', '"')  # noqa: E501
+
+    # Make dictionary of table_name: table_properites. Without whitespaces. For example:
+    #   "ens_mappings": "CREATETABLEIFNOTEXISTSens_mappings(addressTEXTNOTNULLPRIMARYKEY,ens_nameTEXTUNIQUE,last_updateINTEGERNOTNULL);"  # noqa: E501
+    #
+    tables_data_from_constant = {name: properties for name, properties in re.findall(
```

Comment:
So I see why you do it like this but would it not make more sense to pre-create these strings and save them in a file and have them with the app? And make a test to make sure they are always up to date by us? Otherwise everytime you call this function you are recreating this string.

Now not sure how slow it is, but since it's happening multiple times during the startup process I would prefer to have it as fast as possible since startup should be fast.